src/OpenColorIO/Mutex.h

```diff
@@ -24,18 +25,26 @@ class DebugLock
     DebugLock() = default;
     DebugLock(const DebugLock &) = delete;
     DebugLock& operator=(const DebugLock &) = delete;
-    ~DebugLock() { assert(!m_locked); }
-
-    void lock()     { assert(!m_locked); m_mutex.lock();   m_locked = true;  }
-    void unlock()   { assert(m_locked);  m_mutex.unlock(); m_locked = false; }
-    void try_lock() { assert(!m_locked); m_locked = m_mutex.try_lock(); }
-
-    bool locked() const { return m_locked; }
+    ~DebugLock() { assert(m_owner == std::thread::id()); }
+
+    void lock()
+    {
+        assert(m_owner != std::this_thread::get_id());
+        m_mutex.lock();
+        m_owner = std::this_thread::get_id();
+    }
+    void unlock()
+    {
+        assert(m_owner == std::this_thread::get_id());
+        m_owner = std::thread::id();
+        m_mutex.unlock();
+    }
+    bool try_lock() { return m_mutex.try_lock(); }
```

Comment:
Should the `try_lock()` implementation also have the `assert(m_owner != std::this_thread::get_id());` line?
I think it should.

Reply:
Yes, seems it's undefined behavior to `try_lock()` when already holding the lock.